### ISSUE TITLE
Packing was not performed. Packer has exited with code 0xFFFFFFFF. Unknown error.

### ISSUE THREAD

**Issue by Vanir-Labs** [SerGreen/Appacker#28]
I was trying to pack a set of files which the main.exe relies on to function. These file types are very niche and likely unheard of by most. Would this have a reason to do with the errors presented? 

File size is estimated to be about 8GB and the output of the application is under 6MB.

Please let me know what else I can provide to resoluve this issue. This application is amazing work.

EDIT: I tested it more and it seems to refuse to pack my 'grf' file type.

**Comment by SerGreen** [comment]
No, file type should make no difference on the packing process, it doesn't analyze anything, just stuffs all the files inside the package.  

However, it is known that Windows Defender (and other antivirus software) often flags Appacker and its resulting packages as malware due to the way Appacker works (extracting other exe files from itself and executing them), and it's possible that Windows Defender may flag and delete an important file right during the process of packaging. If that happens you will get such error, so maybe that's what's happening here. Check Windows notifications and Defender log to see if it has blocked Appacker files. In Defender log you can then allow and restore these files and it should fix the error. Restart Appacker after that.  

Files that Defender may block are `Appacker.exe` itself and files `packer.exe` and `unpacker.exe` inside a temp directory (path looks like this: `C:\Users\username\AppData\Local\Temp\be33fc68-91b0-464f-a51d-796ea8759f58\unpacker.exe` - temp dir name is random).

**Comment by Vanir-Labs** [comment]
So I got one of the files to pack inside. It was less than 300mb though, I couldn't get larger files into the packer. Is there a way to increase how big a file can be? I have a 9GB file.

**Comment by SerGreen** [comment]
>I have a 9GB file.  

Holy cow... I get it now. Appacker is probably a 32 bit application (i never explicitly changed anything to make it 64 bit) and with 32 bits the maximum file size an app can use is 4 GB.  

Yes, i just checked, it's a 32 bit executable.  

Honestly, i never thought anyone would pack such huge files with it. It should be possible to fix this issue by recompiling Appacker as a 64 bit application. But are you sure you want to create a package this big? Note that every time you launch packed application it extracts a copy of all packed files into a temp directory and then deletes them when you close the packed app. So a package this big will write 9+ GB of data into Temp with every launch. Make sure that a simple archive is not a better option for your task.  

In any case, i will recompile and upload a 64 bit version of Appacker some time soon. Thanks for reporting this issue.

**Comment by Vanir-Labs** [comment]
Thanks for the response. I managed to get my files down to 4GB. Is there a way I can have, like maybe 2 or 3 files not be deleted each time they are loaded? I have two files now about 4.5 GB each in size which only need to be loaded once since they come with encryption. The rest is just DLLs and exes that can be loaded each time.
I have an issue with players hacking the client and exploiting the DLLs when they are not removed. If I can delete them every time and replace them with a fresh one on launch, that would help. This is why I ask.

**Comment by SerGreen** [comment]
Hm, players? So you want to use it to distribute a game? It may be not a good idea because of how easily anti-malware software gets triggered by Appacker. Even packed apps may be flagged as a malware.  

But to answer the question, it's kind of possible to achieve what you want, but it's sub-optimal. Right now the best option would be to disable "Self-repackable application" checkbox in Advanced options (this will make sure that changes to files are not saved between launches) and select Unpack directory as "Same directory as the packed app executable". However, this means that a temp folder will be created in the same directory and all packed files will go into that temp folder, not right next to the packed exe, which is probably what would be preferable for the task. Here's what i mean:  
![image](https://github.com/SerGreen/Appacker/assets/4000373/960b8f85-5bd2-4e69-aeec-d9de4bbcd261)

If your game executable is looking for those other big files **in the parent directory** then it will work. If big files have to be next to unpacked exe then it's currently not possible to achieve. It would be neat to add such functionality to Appacker, though.

**Comment by SerGreen** [comment]
Reporting that simply recompiling as 64 bit application did not help, it crashes because of how files are processed. It tries to read entire file into memory first and it runs out of it if a file is too big. So to fix this issue it will require to change the way packing works.

**Comment by hypy33** [comment]
Hello, i need update too :) SerGreen i have proposition for you support@pulsarinbox.com / discord : _hypy
